### ISSUE TITLE
XML file issue - numerous resources

### ISSUE THREAD

**Issue by monika-aashe** [AASHE/hub#575]
Similar issue to #548 
The PDF download results in an XML file issue. A user was requesting it, and I was unable to find it via search on the institution's website. After digging in, it appears that the XML issue is impacting numerous resources. I didn't check every content type, but so far, it looks like all uploaded files under Green Funds and Green Power Projects have the error. Here are the links to some of the ones I have found:

Green Funds:
https://hub.aashe.org/browse/greenfund/21059/Student-Sustainability-Fund
https://hub.aashe.org/browse/greenfund/21052/EcoFund
https://hub.aashe.org/browse/greenfund/21086/Vanderbilt-Green-Fund-VGF
https://hub.aashe.org/browse/greenfund/21088/Green-Fund

Green Power:
https://hub.aashe.org/browse/greenpowerproject/17425/545-kW-Rooftop-Array-Recreation-Center
https://hub.aashe.org/browse/greenpowerproject/17512/220-kW-Rooftop-Array-Veterinary-Teaching-Hospital
https://hub.aashe.org/browse/greenpowerproject/17546/139-kW-Rooftop-Array-Parmalee-Hall

Thanks in advance for your help on this!

**Comment by rerb** [comment]
Possible solution:  rather than linking to actual filename (which can contain characters that are illegal in URLS, like spaces, and pluses, I guess), link to a url that contains the urlencoded version of the file name (i.e., urlencode(file.name) or something similar).

**Comment by rerb** [comment]
might be as easy as a change to a template?  but which one . . .

**Comment by monika-aashe** [comment]
This is also happening with at least one conference presentation. I can download it in the Admin, but if I try to download it and re-upload in the Admin, the XML file issue persists. 

https://hub.aashe.org/browse/presentation/15993/learn-how-to-launch-a-university-surplus-program-and-start-making-real-money-for-sustainability-while-reducing-your-waste

**Comment by rerb** [comment]
Could the problem be uploading through the admin?  Does the Hub web UI tweak (urlencode) the URL it targets, a bit of processing we might be short-circuit when upLoading through Django admin app?  If you upload through the hub app, rather than the admin, does the problem persist?

—
Bob

> On Oct 4, 2019, at 1:37 PM, monika-aashe <notifications@github.com> wrote:
> 
> ﻿
> This is also happening with at least one conference presentation. I can download it in the Admin, but if I try to download it and re-upload in the Admin, the XML file issue persists.
> 
> https://hub.aashe.org/browse/presentation/15993/learn-how-to-launch-a-university-surplus-program-and-start-making-real-money-for-sustainability-while-reducing-your-waste
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.

-- 
Join us in beautiful Spokane, Washington, US, for the 2019 AASHE Conference 
& Expo, on October 27-30!  Register now at http://www.aashe.org/conference 
<http://www.aashe.org/conference>.


Connect with us:   Website 
<http://www.aashe.org/> | Facebook <https://www.facebook.com/aasheorg> | 
Twitter <https://twitter.com/AASHENews> | LinkedIn 
<http://www.linkedin.com/company/aashe>

**Comment by monika-aashe** [comment]
Bob, it appears that, at least for the MOST RECENT, issue, the problem is due to the characters in the file name (plus signs). When I uploaded separately under the same file name with the pluses, the same issue occurred. When I edited the file name to include hyphens instead of plus signs, it uploaded fine. Here is a test (I will delete it soon, so it may not be there if you look at it later):

https://hub.aashe.org/browse/presentation/22255/test-1

**Comment by monika-aashe** [comment]
However, this does not explain the issue of why all Green Funds and Green Power Project uploads have the same error. The files I've looked at there don't seem to have unusual characters.

**Comment by juliandsmith** [comment]
@rerb I know you have a lot on your plate but Monika and I are wondering if you have a sense of when you might be able to look into (and hopefully) fix this issue.

**Comment by dodget** [comment]
@monika-aashe - Do you still have the green funds spreadsheet that I used to import the content? I no longer have access to it, but it would be nice to have for debugging. If it's a google sheet, you can share it with tylor.dodge1@gmail.com.

**Comment by dodget** [comment]
I have it actually. Sorry, just remembering where things are!

**Comment by dodget** [comment]
Ok, so there were two things happening here. The first is an issue that I just detailed in #577. The second has to do with the Green Power Projects import as well as the Green Funds import.

Regarding the second reason, both of those import processes used a function to upload files from urls to S3. That process failed to successfully download the file from the url, so it just uploaded nothing. To resolve this I went through the csv files that were used for importing content in those two processes, and I manually uploaded any files that were available. If the url of the file to be uploaded produced a `Page Not Found`, then I deleted "file" on the resource, because there was nothing to upload.

This isn't a problem generally for Conference Presentations, because in that process I manually upload files to S3, and then associate the resource with a file that is already uploaded. If there are some issues with other resources in general, I'd say it has to do with #577 which I just opened.

I recommend that this issue be closed, as I fixed resources from the Green Power Projects and Green Funds uploads, and that further issues be tracked in #577.
